assets/js/romo/modal.js

```diff
@@ -110,153 +220,74 @@ RomoModal.prototype.doResetBody = function() {
-  if (e !== undefined) {
-    e.preventDefault();
-  }
-
-  if ((Romo.hasClass(this.elem, 'disabled') === false) &&
-      (Romo.hasClass(this.popupElem, 'romo-modal-open') === false)) {
+RomoModal.prototype._onPopupOpen = function(e) {
+  if (Romo.hasClass(this.elem, 'disabled') === false && this.popupClosed()) {
     setTimeout(Romo.proxy(function() {
       this.doPopupOpen();
-    }, this), 100);
-  }
-}
-
-RomoModal.prototype.doPopupOpen = function() {
-  if (Romo.data(this.elem, 'romo-modal-content-elem') !== undefined) {
-    var contentElem = Romo.elems(Romo.data(this.elem, 'romo-modal-content-elem'))[0];
-    this.doLoadBodySuccess(contentElem.outerHTML);
-  } else {
-    this.romoAjax.doInvoke();
+    }, this), 1);
   }
-
-  Romo.addClass(this.popupElem, 'romo-modal-open');
-  this.doPlacePopupElem();
-
-  // bind an event to close the popup when clicking away from the
-  // popup.  Bind on a timeout to allow time for any toggle
-  // click event to propagate.  If no timeout, we'll bind this
-  // event, then the toggle click will propagate which will call
-  // this event and immediately close the popup.
-  setTimeout(Romo.proxy(function() {
-    this.doBindWindowBodyClick();
-  }, this), 100);
-
-  // bind "esc" keystroke to toggle close
-  this.doBindWindowBodyKeyUp();
-
-  // bind window resizes reposition modal
-  Romo.on(window, 'resize', Romo.proxy(this.onResizeWindow, this));
-
-  Romo.trigger(this.elem, 'romoModal:popupOpen', [this]);
 }
 
-RomoModal.prototype.onPopupClose = function(e) {
-  if (e !== undefined) {
-    e.preventDefault();
-  }
-
-  if (Romo.hasClass(this.elem, 'disabled') === false) {
+RomoModal.prototype._onPopupClose = function(e) {
+  if (Romo.hasClass(this.elem, 'disabled') === false && this.popupOpen()) {
     setTimeout(Romo.proxy(function() {
       this.doPopupClose();
-    }, this), 100);
+    }, this), 1);
   }
 }
 
-RomoModal.prototype.doPopupClose = function() {
-  Romo.trigger(Romo.f('body')[0], 'romoModal:popupclose');
-  Romo.removeClass(this.popupElem, 'romo-modal-open');
-
-  // unbind any event to close the popup when clicking away from it
-  this.doUnBindWindowBodyClick();
-
-  // unbind "esc" keystroke to toggle close
-  this.doUnBindWindowBodyKeyUp();
-
-  // unbind window resizes reposition modal
-  Romo.off(window, 'resize', Romo.proxy(this.onResizeWindow, this));
-
-  // clear the content elem markup if configured to
-  if (Romo.data(this.elem, 'romo-modal-clear-content') === true) {
-    Romo.updateHtml(this.contentElem, '');
-  }
-
-  Romo.trigger(this.elem, 'romoModal:popupClose', [this]);
-}
-
 RomoModal.prototype.onMouseDown = function(e) {
-  e.preventDefault();
-  e.stopPropagation();
-  this.doDragStart(e);
+  this._dragStart(e);
   return false;
 }
 
-RomoModal.prototype.doDragStart = function(e) {
+RomoModal.prototype._dragStart = function(e) {
   this.dragElem.addClass('romo-modal-grabbing');
   this.dragElem.removeClass('romo-modal-grab');
```

Comment:
@kellyredding - Be sure to update these as well; thanks for fixing what I missed.